src/toil/batchSystems/htcondor.py

```diff
@@ -0,0 +1,306 @@
+                        schedd_name, condor_host))
+                try:
+                    schedd_ad = htcondor.Collector(condor_host).locate(
+                        htcondor.DaemonTypes.Schedd, schedd_name)
+                except IOError:
+                    logger.error(
+                        "Could not connect to HTCondor Collector at {0}".format(condor_host))
+                    raise
+                except ValueError:
+                    logger.error(
+                        "Could not find HTCondor Schedd with name {0}".format(schedd_name))
+                    raise
+                else:
+                    schedd = htcondor.Schedd(schedd_ad)
+
+            # Otherwise assume the Schedd is on the local machine
+            else:
+                logger.debug("Connecting to HTCondor Schedd on local machine")
+                schedd = htcondor.Schedd()
+
+            # Ping the Schedd to make sure it's there and responding
+            try:
+                schedd.xquery(limit = 0)
+            except RuntimeError:
+                logger.error("Could not connect to HTCondor Schedd")
+                raise
+
+            return schedd
+
+        def getEnvString(self):
+            '''Build an environment string that a HTCondor Submit object can use.
+
+            For examples of valid strings, see:
+            http://research.cs.wisc.edu/htcondor/manual/current/condor_submit.html#man-condor-submit-environment
+
+            '''
+
+            env_items = []
+            if self.boss.environment:
+                for key, value in self.boss.environment.items():
+                    
+                    # Each variable should be in the form of <key>='<value>'
+                    env_string = key + "="
+
+                    # The entire value should be encapsulated in single quotes
+                    # Quote marks (single or double) that are part of the value should be duplicated
+                    env_string += "'" + value.replace("'", "''").replace('"', '""') + "'"
+
+                    env_items.append(env_string)
+
+            # The entire string should be encapsulated in double quotes
+            # Each variable should be separated by a single space
+            return '"' + ' '.join(env_items) + '"'
+
+    # Override the issueBatchJob method so HTCondor can be given the disk request
+    def issueBatchJob(self, jobNode):
+        # Avoid submitting internal jobs to the batch queue, handle locally
+        localID = self.handleLocalJob(jobNode)
+        if localID:
+            return localID
+        else:
+            self.checkResourceRequest(jobNode.memory, jobNode.cores, jobNode.disk)
+            jobID = self.getNextJobID()
+            self.currentJobs.add(jobID)
+
+            # Add the jobNode.disk and jobNode.jobName to the job tuple
+            self.newJobsQueue.put((jobID, jobNode.cores, jobNode.memory, jobNode.disk, jobNode.jobName, jobNode.command))
```

Comment:
General question for Toil devs, why don't we put the `jobNode` itself in the `newJobsQueue`?

Reply:
Toil only needs the jobId. It seems the queue was only jobIds originally and later changed to a tuple to include extra information used by the batch system.